flink-runtime/src/test/java/org/apache/flink/runtime/rest/RestClientTest.java

```diff
@@ -84,6 +89,26 @@ public void testConnectionTimeout() throws Exception {
         }
     }
 
+    @Test
+    public void testHttpsConnectionWithDefaultCerts() throws Exception {
+        final Configuration config = new Configuration();
+        final URL httpsUrl = new URL("https://raw.githubusercontent.com");
```

Comment:
Dependency on that endpoint is not ideal for a unit test. However, given that the CI setup already won't function w/o that connection and the difficulty to test this specific functionality (HTTPS with default keystore), I think this exception is justifiable. I would suggest though to add a comment though for the unlikely case that someone runs tests locally and is not connected to the internet.

Reply:
> given that the CI setup already won't function w/o that connection

This was exactly what I was thinking, however it is a very good point about the local Unit tests execution. It is not optimal to put someone into the position when they cannot run unit tests without an internet connection. How about we move it into a Java-based E2E test (in `flink-end-to-end-tests`), similar to `KafkaSourceE2ECase`. It will only be executed with the `run-end-to-end-tests` profile enabled (in CI). I am a bit not sure about the appropriate module name here. Maybe `flink-restclient-test` ?